doc/tutorials/core/univ_intrin/univ_intrin.markdown

```diff
@@ -0,0 +1,269 @@
+
+* **Load**: Load functions allow you to *load* values into a register.
+    * *Constructors* - When declaring a register structure, we can either provide a memory address from where the register will pick up contiguous values, or provide the values explicitly as multiple arguments (Explicit multiple arguments is available only for Constant Sized Registers):
+
+        ```
+        float ptr[32] = {1, 2, 3 ..., 32};
+        // ptr is a pointer to a contiguous memory block of 32 floats
+
+        // Variable Sized Registers //
+        int x = v_float32().nlanes; // set x as the number of values the register can hold
+
+        v_float32 reg1(ptr);      // reg1 stores first x values according to the maximum register size available.
+        v_float32 reg2(ptr + x);  // reg stores the next x values
+
+
+
+        // Constant Sized Registers //
+        v_float32x4 reg1(ptr);        // reg1 stores the first 4 floats (1, 2, 3, 4)
+        v_float32x4 reg2(ptr + 4);    // reg2 stores the next 4 floats (5, 6, 7, 8)
+
+        // Or we can explicitly write down the values.
+        v_float32x4(1, 2, 3, 4);
+        ```
+
+    * *Load Function* - We can use the load method and provide the memory address of the data:
+
+        ```
+        float ptr[32] = {1, 2, 3, ..., 32};
+        v_float32 reg_var;
+        reg_var = vx_load(ptr);    // loads values from ptr[0] upto ptr[nlanes - 1]
+
+        v_float32x4 reg_128;
+        reg_128 = v_load(ptr);     // loads values from ptr[0] upto ptr[3]
+
+        v_float32x8 reg_256;
+        reg_256 = v256_load(ptr);  // loads values from ptr[0] upto ptr[7]
+
+        v_float32x16 reg_512;
+        reg_512 = v512_load(ptr);  // loads values from ptr[0] upto ptr[15]
+        ```
+
+
+        @note The load function assumes data (a[] in this case) is unaligned. If your data is aligned, you may use the @ref vx_load_aligned function.
+
+
+* **Store**: Store functions allow you to *store* the values from a register into a particular memory location.
+    * To store values from a register into a memory location, you may use the @ref v_store() function:
+
+        ```
+        float ptr[4];
+        v_store(ptr, reg); // store the first 128 bits(interpreted as 4x32-bit floats) of reg into ptr.
+        ```
+
+        @note reg containing values of a type may lead to wrong interpretation during storage. For example: if **ptr** is a pointer to a 32-bit float and reg is a register containing 32-bit integers,  the values in ptr may not be as intended. You may cast the registers into the proper type before carrying out operations.
+
+#### Binary and Unary Operators
+
+The universal intrinsics set provides element wise binary and unary operations.
+
+* **Arithmetics**: We can add, subtract, multiply and divide two registers element-wise. The registers must be of the same width and hold the same type. To multiply two registers, for example:
+
+    ```
+    v_float32x4 a, b;   // {a1, a2, a3, a4}, {b1, b2, b3, b4}
```

Comment:
Could you please use Wide Universal Intrinsics approach for all operations except loading. WUI approach make the code suitable for any platform regardless of the exact register size and so is the preferred way to use universal intrinsics. Fixed register width approach should be used for really specific cases(which are definitely outside of the basic tutorial scope) and usually such implementations in the library code are a legacy.